dotCMS/src/main/java/com/dotmarketing/filters/CMSUrlUtil.java

```diff
@@ -530,4 +532,14 @@ public String getQueryStringFromUri(final String uri) {
 				uri.substring(indexOf+1):
 				null;
 	}
+
+	public static String getCurrentURI(final HttpServletRequest request)  {
+		try {
+			return URLDecoder.decode((request.getAttribute(Constants.CMS_FILTER_URI_OVERRIDE) != null)
+					? (String) request.getAttribute(Constants.CMS_FILTER_URI_OVERRIDE)
+					: request.getRequestURI(), "UTF-8");
+		} catch (UnsupportedEncodingException e) {
+			throw new DotRuntimeException(e);
```

Comment:
is it worth to add a `Log.debug` ? or warning?

Reply:
done https://github.com/dotCMS/core/pull/17859/commits/c99f44a7598559801ecaf90b4041c07a374915e6#diff-b01fee3e8841e6997654dbe68e5baf2dR542